File: docs/snippets/Docs.fs
module Docs

/// <summary>
/// Print a decoding result: the value on success, the error message as Thoth.Json wrote it.
/// </summary>
let print (result: Result<'T, string>) =
    match result with
    | Ok value -> printfn "%A" value
    | Error error -> printfn "%s" error

File: assets/live-example/precompiled/e667cbc66489906b/Docs.js

import { printf, toConsole } from "fable-library-js/String.js";

/**
 * Print a decoding result: the value on success, the error message as Thoth.Json wrote it.
 */
export function print(result) {
    if (result.tag === 1) {
        toConsole(printf("%s"))(result.fields[0]);
    }
    else {
        toConsole(printf("%A"))(result.fields[0]);
    }
}

